### DISCUSSION TITLE
How can I generate or find a file for a module for my car?

### DISCUSSION THREAD

**Discussion by vlovini** [cedricp/ddt4all#1176] [General]
Hi colleagues!

First of all, congratulations on this impressive project. I will see if I can contribute myself in the future :)

I just bought a Sandero RS (Sandero II in its brazilian hot hatch version) and want to activate some things that I'm missing from my previous car (I came from a Ford, so I'm used to the Forscan softwares and modules).

How can I find or generate configuration files for some modules of my car? Being more specifically, I need a USM External lighting module that didn't match with any of the ones available in the ecu.zip that I found on the internet.

Is it possible to extract this from the car or from a maintenance site?

Note: Sorry if the question sound a little bit dumb, but I'm new in the Renault community :)

Thanks in advance!

**Comment by vlovini** [comment]
Also: I'm trying to enable Welcome/Goodbye lights for my car as one of the first things. If somebody made that for the Sandero II or Duster, you may give some hints form me (I've saw that the Kwid is able to enable that).

**Reply by AndyIreland** [reply]
which EMM ECU has your car got though , what does it find when you do a search?
Your ECU might not even have the welcome and goodbye feature on it depending on year and which EMM is fitted to your car. 

Plus what year is a Sandero II ? - if its a late year around 2020 car it might be protected with security gateway that needs unlocking token before changes can be made?
![image](https://github.com/user-attachments/assets/0fece408-e0e8-455f-903c-981e9f75f66b)

**Reply by vlovini** [reply]
Hi Andy! Found your video on youtube yesterday :)

Thanks for the fast response!

It is a 2019/2020 Sandero II. I actually don't know which EMM ECU my car has, can you teach me how to find out?

About the security gateway - how can I proceed with that?

Note: I'm using the ported version from the DDT4ALL app on a Android device (ECU tweaker). ASAP I will format my laptop that is running linux to install the Windows app.

**Reply by AndyIreland** [reply]
Hi there . Its not going to be easy. Yep off the top of my head I reckon your OBD2 socket has got security gateway being that year of car. - You have to get it unlocked at a renault main dealer or garage that can order up a 'Token' to unlock the OBD2 system so to change parameters on the cars ECU's.
There are some people claim they have 'bypassed' this gateway by wiring up directly to BCM wiring behind the dashboard but I dont know to be honest. -  there is this thread on Clio Mk5
https://github.com/cedricp/ddt4all/discussions/956
 , I know you have a Dacia but its same kind of thing , a lot of reading at over 500 comments but just skip to the bits about bypassing gateway.
I dont have enough use on the ECU Tweaker (DDT4ALL Android ported) but using a laptop you 'search' for your ECU's in your car using the magnifying glass 
![image](https://github.com/user-attachments/assets/a76380b2-9f67-4647-9373-eeb9631bfadc)
 in DDT4ALL (Laptop version)

**Reply by vlovini** [reply]
If this firewall is a thing on my car, shouldn't it block me to read/write all the ECU modules?
Because I'm able to read and write BCM options. I just can't confirm with I'm having a firewall or a file definition in regards of the EMM module.

**Reply by AndyIreland** [reply]
true - if you have got SGW by rights you shouldnt be able to change any parameters , so yeah they introduced it around 2020 time , maybe your car just missed it or was produced earlier say 2019 but sold it 2020

**Reply by vlovini** [reply]
![modules](https://github.com/user-attachments/assets/5ada648f-4aa3-4d18-a3fd-a67b9be75230)
Just tried using the computer. I think I don't have good news. These are the only modules available via scan (besides other two regarding the engine that is not in this screenshot). 

Am I missing something? I have another ELM327 scanner that I could try (an USB one with MS/HS switch that I used to use for my previous Ford car. Is this MS/HS switch a thing for Renault/Dacia?)

Is there something more I could do?

Thanks in advance again!

**Reply by AndyIreland** [reply]
i dont think your ford one with switch HS/MS will do ,  thats for FORSCAN I think, High and low 

CAN on Renault wiring different to FORSCAN MS/HS

Make sure when using DDT4ALL you are using an ELM327 with a PIC18F25K80 chip in it for best compatibility and try avoid cheap china clone ELM327  

Try also increasing CAN timeout to 600ms 

other than that - It strange it just recognise BCM (body control module) but not the others . Items in red I think mean caution not exact ECU match so proceed with care adjusting parameters on these ones 
Green ones in box normally mean exact match in the database.

**Comment by matzel82** [comment]
Sometimes I really ask me…what you are doing with your cars….

**Reply by AndyIreland** [reply]
if that's directed at me ... I am a tinkerer :)

**Comment by matzel82** [comment]
> Hi colleagues!
> 
> First of all, congratulations on this impressive project. I will see if I can contribute myself in the future :)
> 
> I just bought a Sandero RS (Sandero II in its brazilian hot hatch version) and want to activate some things that I'm missing from my previous car (I came from a Ford, so I'm used to the Forscan softwares and modules).
> 
> How can I find or generate configuration files for some modules of my car? Being more specifically, I need a USM External lighting module that didn't match with any of the ones available in the ecu.zip that I found on the internet.
> 
> Is it possible to extract this from the car or from a maintenance site?
> 
> Note: Sorry if the question sound a little bit dumb, but I'm new in the Renault community :)
> 
> Thanks in advance!

What mean…“you miss something“

**Reply by vlovini** [reply]
I'm just starting on this Renault/Dacia world...
Some of the things I missed already:
- Window lift (already activated)
- Welcome/Goodbye lights
- Follow me home lights

For the moment that's it. Still finding out what else is possible...

**Reply by AndyIreland** [reply]
I never managed to get window lift working on our Clio mk4 and renault Zoe - i reckon it needs factory fitted Renault alarm to get it to work 

i Got welcome & GoodBye lights working on Clio4 .... but then it put the DRL's at 100% brightness all the time, even when headlights were on (where they dimmed or whent out totally before) 

Follow me home lights worked as standard on the clio4 and Zoe - I didnt have to programme that

**Comment by vlovini** [comment]
Hi Andy,

Tried out with a better ELM237 and timeout increased to 1000ms but had no
luck.

![image](https://github.com/user-attachments/assets/18855470-d6c7-4d8f-954e-89ecd31f70d4)


Other modules appeared, such as the ABS one, but nothing about the EMM.
I guess this car has no special configuration for lights...

**Reply by AndyIreland** [reply]
maybe one day you can try a ELM327 with a PIC18F25K80 and see if it finds all. 
there is even a way in DDT4ALL you can try manually selecting an ECU from an xml file - it can be dangerous though I think , you could end up rendering your car (or your cars lighting say if you are changing lighting parameters on the lighting ecu) useless.

**Reply by Furtif** [reply]
Sim isso https://github.com/cedricp/ddt4all?tab=readme-ov-file#supported-diagnostic-adapters-so-far ler ai, nem todos obd são bons.

**Reply by vlovini** [reply]
Hi guys,

As far as I understand, I'm using a good one. 

It uses a pic 18f25k80 as well. 
![SmartSelect_20240726_112911_AliExpress](https://github.com/user-attachments/assets/52bf10b7-6dda-435a-a888-dc45e277fd8c)
